### ISSUE TITLE
Multiple digitcaps layers generate nan training loss

### ISSUE THREAD

**Issue by Ralphyan** [XifengGuo/CapsNet-Keras#43]
When I add a second digitcaps layer, the training process starts to generate nan loss. What causes this problem ?  Thank you so much
In capsulenet.py, I changed the following code from:

digitcaps = CapsuleLayer(num_capsule=n_class, dim_capsule=16, routings=routings,
                             name='digitcaps')(primarycaps)

to 

digitcaps2 = CapsuleLayer(num_capsule=n_class, dim_capsule=16, routings=routings,
                             name='digitcaps2')(primarycaps)
digitcaps = CapsuleLayer(num_capsule=n_class, dim_capsule=16, routings=routings,
                             name='digitcaps')(digitcaps2)

**Comment by XifengGuo** [comment]
@Ralphyan It's due to too small value of initialized W. You can solve this by replacing the following line
https://github.com/XifengGuo/CapsNet-Keras/blob/540f15dfc531c3e82f76fbae7de72bb9e48ae72a/capsulelayers.py#L97
with `self.kernel_initializer = initializers.random_uniform(-1, 1)`
Of cause, you can try other initialization methods.
